frontend/packages/console-app/src/actions/providers/storageclass-provider.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import i18next from 'i18next';
+import { Action } from '@console/dynamic-plugin-sdk';
+import { useK8sWatchResource } from '@console/dynamic-plugin-sdk/src/api/dynamic-core-api';
+import {
+  getGroupVersionKindForModel,
+  k8sPatchResource,
+} from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { defaultClassAnnotation } from '@console/internal/components/storage-class';
+import { asAccessReview } from '@console/internal/components/utils';
+import { K8sResourceCommon, K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
+import { useK8sModel } from '@console/shared/src/hooks/useK8sModel';
+import { getCommonResourceActions } from '../creators/common-factory';
+
+export const useStorageclassActions = (
+  storageClass: K8sResourceKind,
+): [Action[], boolean, boolean] => {
+  const [storageClassModel, inFlight] = useK8sModel(referenceFor(storageClass));
+
+  const [storageClasses] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: getGroupVersionKindForModel(storageClassModel),
+    isList: true,
+  });
+
+  const existingDefaultStorageClass = React.useMemo(
+    () =>
+      storageClasses.find((sc) => sc.metadata?.annotations?.[defaultClassAnnotation] === 'true'),
+    [storageClasses],
+  );
+
+  const isDefaultStorageClass =
+    storageClass.metadata?.annotations?.[defaultClassAnnotation] === 'true';
+
+  const storageClassActions: Action[] = React.useMemo(
+    () => [
+      ...getCommonResourceActions(storageClassModel, storageClass),
+      {
+        accessReview: asAccessReview(storageClassModel, storageClass, 'patch'),
+        disabled: isDefaultStorageClass,
+        disabledTooltip: isDefaultStorageClass ? i18next.t('Current default StorageClass') : null,
```

Comment:
lets use the `useTranslation()` here instead, passing the namespace as well. As you may notice the generated string is in the wrong package.